channeldb/graph.go

```diff
@@ -439,18 +446,33 @@ func (c *ChannelGraph) SetSourceNode(node *LightningNode) error {
 	})
 }
 
+// BatchedAddLightningNode adds a vertex/node to the graph database. If the node
+// is not in the database from before, this will add a new, unconnected one to
+// the graph. If it is present from before, this will update that node's
+// information. Note that this method is expected to only be called to update an
+// already present node from a node announcement, or to insert a node found in a
+// channel update.
+func (c *ChannelGraph) BatchedAddLightningNode(node *LightningNode) error {
+	return c.nodeScheduler.Execute(&batch.Request{
+		Update: func(tx *bbolt.Tx) error {
+			return addLightningNode(tx, node)
+		},
+		Solo: func() error {
+			return c.AddLightningNode(node)
+		},
+	})
+}
+
 // AddLightningNode adds a vertex/node to the graph database. If the node is not
 // in the database from before, this will add a new, unconnected one to the
 // graph. If it is present from before, this will update that node's
-// information. Note that this method is expected to only be called to update
-// an already present node from a node announcement, or to insert a node found
-// in a channel update.
+// information. Note that this method is expected to only be called to update an
+// already present node from a node announcement, or to insert a node found in a
+// channel update.
 //
 // TODO(roasbeef): also need sig of announcement
 func (c *ChannelGraph) AddLightningNode(node *LightningNode) error {
```

Comment:
at some point, yes possibly. currently they are still useful esp. in unit tests, since we wouldn't have to set custom timeouts and stuff for tests. it would be a pretty big investment to do so tho because most of the test infra is already written with the one-off methods

Reply:
ended up making the timeout configurable from the config file, so everything passes through the original methods!